File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1074_basico.js
import { question } from "readline-sync";

function main() {
    const N = get_number();
    if (N_Valido(N)) {
        let contador = 1;

        while (contador <= N) {
            let X = get_number();
            if (X_Valido(X)) {
                if (X > 0) {
                    X = (X % 2 === 0) ? `EVEN POSITIVE` : `ODD POSITIVE`;
                } else if (X < 0) {
                    X = (X % 2 === 0) ? `EVEN NEGATIVE` : `ODD NEGATIVE`;
                } else {
                    X = `NULL`
                }
                print(`${X}`);
                contador++;
            }
        }
    }
}

function get_number() {
    return parseInt(question());
}

function print(texto) {
    return console.log(texto);
}

function N_Valido(numero) {
    return (numero < 10_000);
}

function X_Valido(numero) {
    return (numero > -(10 ** 7) && numero < (10 ** 7));
}

main();

File: Exercicios_ED_I/Bee/1074_basico.cpp
#include <stdio.h>
#include <iostream>
using namespace std;

main() {
    int n, i = 0;
    cin >> n;

    while (i < n) {
        int numero;
        cin >> numero;

        if (numero == 0) {
            cout << "NULL" << endl;
        } else if (numero % 2 == 0) {
            if (numero > 0) {
                cout << "EVEN POSITIVE" << endl;
            } else {
                cout << "EVEN NEGATIVE" << endl;
            }
        } else {
            if (numero > 0) {
                cout << "ODD POSITIVE" << endl;
            } else {
                cout << "ODD NEGATIVE" << endl;
            }
        }
        i++;
    }
}